fiat/prices.go

```diff
@@ -63,9 +66,17 @@ type PriceSourceConfig struct {
 // given the chosen price backend.
 func (cfg *PriceSourceConfig) validatePriceSourceConfig() error {
 	switch cfg.Backend {
-	case UnknownPriceBackend, CoinCapPriceBackend:
+	// If an unknown price backend is set, we don't expect granularity to
+	// be set. The caller should provide a backend if they wish to use
+	// a specific granularity.
+	case UnknownPriceBackend:
```

Comment:
or coinDesk?

Reply:
Nice catch!

Added a separate case for coindesk to allow setting daliy granularity, since that's what it provides, but failing otherwise + updated unit tests.